src/runtime/graph/graph_runtime.cc

```diff
@@ -103,27 +99,55 @@ class GraphRuntime : public ModuleNode {
   void SetInput(int index, DLTensor* data_in) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(data_in, &data_entry_[eid], nullptr));
+    data_entry_[eid].CopyFrom(data_in);
   }
   /*!
-   * \brief Copy index-th input to data_out
+   * \brief Get the number of outputs
+   *
+   * \return The number of outputs from graph.
+   */
+  int NumOutputs() const {
+    return outputs_.size();
+  }
+  /*!
+   * \brief Return NDArray for given input index.
    * \param index The input index.
-   * \param data_out The output
+   *
+   * \return NDArray corresponding to given input node index.
    */
-  void GetInput(int index, DLTensor* data_out) {
+  NDArray GetInput(int index) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(&data_entry_[eid], data_out, nullptr));
+    return data_entry_[eid];
+  }
+  /*!
+   * \brief Return NDArray for given output index.
+   * \param index The output index.
+   *
+   * \return NDArray corresponding to given output node index.
+   */
+  NDArray GetOutput(int index) {
+    CHECK_LT(static_cast<size_t>(index), outputs_.size());
+    uint32_t eid = this->entry_id(outputs_[index]);
+    return data_entry_[eid];
   }
   /*!
    * \brief Copy index-th output to data_out.
    * \param index The output index.
    * \param data_out the output data.
    */
-  void GetOutput(int index, DLTensor* data_out) {
+  void CopyOutputTo(int index, DLTensor* data_out) {
     CHECK_LT(static_cast<size_t>(index), outputs_.size());
     uint32_t eid = this->entry_id(outputs_[index]);
-    TVM_CCALL(TVMArrayCopyFromTo(&data_entry_[eid], data_out, nullptr));
+
+    // Check the shapes to avoid receiving in different dimension but same size.
```

Comment:
This check breaks some tutorials (from_keras.py, from_mxnet.py, ...)

https://github.com/dmlc/tvm/blob/01ec533e7cfb603f6114a763f03dad4c564f589d/tutorials/nnvm/from_keras.py#L95

I will send a patch later. But I wonder why CI cannot detect this.